First stable version

cambierr opened the pull request:
Highly based on @FokkeZB's work with node.js

FokkeZB commented on README.md:
```diff
@@ -0,0 +1,65 @@
+public class Test {
+
+    public static void main(String[] args) throws MqttException {
+        new Client(Region.STAGING, "MyAppEUI", "MyAppSecret")
```
I think it would be better to keep this a string as the Node.js client has. We'll be adding regions and people will need to be able to use any other host as well.

cambierr replied:
Actually, both are available. you can create a client from _new Client(Region.STAGING, ...)_ and also using _new Client("tcp://staging.thethingsnetwork.org:1883", ...)_

I added this to make it even more easy to use but I can remove that constructor in case you think it's better ?

FokkeZB replied:
I think it's better to not hardcode the regions in the lib so that people will actually consult docs to see what regions are currently available.

Also, I'd like the constructor accept just `staging.thethingsnetwork.org` (without the scheme and port). Is that possible? And if you'd pass a region `eu` it should add `.thethings.network` to it.

FokkeZB commented on src/main/java/org/thethingsnetwork/java/app/lib/Client.java:
```diff
@@ -0,0 +1,323 @@
+     * @return the Client instance
+     * @throws MqttException in case something goes wrong
+     */
+    public Client stopNow() throws MqttException {
```
In the Node.js client this is end(). Might be good to do that here as well?

FokkeZB commented:
Awesome thanks @cambierr, I added a view notes.

cambierr commented:
So I changed the stop() method to end() and stopNow() to endNow().
Also, I renamed the handler registration methods so that lambdas can be used (not sure a lot of people use them but I do, so..)

FokkeZB commented:
@cambierr that sure looks a lot cleaner in the readme, thx.

We're currently setting up a process to sign a CLA. Once that is in place you'd have to sign that before we can merge.

Also, do you have any suggestions on package registries/managers we should submit the library to?

cambierr commented:
The best way to publish it is to do it via maven central since they are available from both maven and gradle.

The pom file of the project is already configured to do so. Just run a _mvn clean package deploy_ and they'll be published to maven (off course, you need to register to sonatype & whitelist org.thethingsnetwork prior to that)

FokkeZB commented:
Thanks @cambierr, we'll publish it to maven central.

Would you also be interested in doing a PR against https://github.com/TheThingsNetwork/docs for the docs? You can skip refactor and just duplicate `_archives/refactor/node-js(.md)`. When you run it locally it'll be available via `/docs/refactor/java`.

cambierr commented:
I'll do that later this week ;-)

johanstokking commented:
Thanks @cambierr !
